OpenRA.Mods.Common/Traits/SpawnActorOnDeath.cs

```diff
@@ -50,16 +50,16 @@ public class SpawnActorOnDeathInfo : ITraitInfo
 			"lead to unexpected behaviour.")]
 		public readonly CVec Offset = CVec.Zero;
 
-		public object Create(ActorInitializer init) { return new SpawnActorOnDeath(init, this); }
+		public override object Create(ActorInitializer init) { return new SpawnActorOnDeath(init, this); }
 	}
 
-	public class SpawnActorOnDeath : INotifyKilled
+	public class SpawnActorOnDeath : ConditionalTrait<SpawnActorOnDeathInfo>, INotifyKilled
 	{
 		readonly SpawnActorOnDeathInfo info;
```

Comment:
Do you mean replacing that line with ```public class SpawnActorOnDeath : ConditionalTrait<ConditionalTraitInfo>, INotifyKilled```? I don't really understand the reason behind it and the interesting thing about it is that it doesn't cause any issues because `SpawnActorOnDeath` has its own field for the info.